store-implementation/federated-store/src/main/java/uk/gov/gchq/gaffer/federatedstore/operation/handler/impl/FederatedRemoveAndDeleteGraphHandler.java

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright 2022 Crown Copyright
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.gchq.gaffer.federatedstore.operation.handler.impl;
+
+import uk.gov.gchq.gaffer.federatedstore.FederatedStore;
+import uk.gov.gchq.gaffer.federatedstore.operation.RemoveGraph;
+import uk.gov.gchq.gaffer.graph.Graph;
+import uk.gov.gchq.gaffer.operation.OperationException;
+import uk.gov.gchq.gaffer.store.Context;
+import uk.gov.gchq.gaffer.store.Store;
+import uk.gov.gchq.gaffer.store.operation.DeleteAllData;
+
+import java.util.Collections;
+import java.util.List;
+
+
+/**
+ * A handler for RemoveGraph operation for the FederatedStore.
+ * And delete associated Accumulo Tables.
+ * <p>
+ *
+ * @see FederatedStore
+ * @see RemoveGraph
+ */
+public class FederatedRemoveAndDeleteGraphHandler extends FederatedRemoveGraphHandler {
+
+    @Override
+    public Boolean doOperation(final RemoveGraph operation, final Context context, final Store store) throws OperationException {
+        try {
+            //Get the graph before they are removed.
+            final List<Graph> graphsToRemove = ((FederatedStore) store).getGraphs(context.getUser(), Collections.singletonList(operation.getGraphId()), operation);
+
+            //Ask graphs to delete all data.
+            for (final Graph graph : graphsToRemove) {
+                if (graph.isSupported(DeleteAllData.class)) {
```

Comment:
If someone has specifically asked for remove **and** delete, might it be useful to warn them if the relevant store doesn't support deletion? Otherwise they might just assume data was deleted when it wasn't because the operation ran successfully.
One option could be to log a warning, but as this might not make its way to the user immediately, you could instead throw an exception if the operation is not supported. This could state that the store doesn't support deleting and therefore the RemoveGraph operation should be used instead.

Reply:
Logging will never make its way to the user.

Reply:
I'm thinking more about awareness for an admin. As it is written currently, if a store doesn't support deleting then the data will be left undeleted and admins would need to know this was a possibility and manually check for this as there's no logging to inform about this.
The easiest option would be to **fail if the store doesn't support deletion**. There's no reason to use this operation over the remove operation if store doesn't support deletion.